internal/server/storage/drivers/driver_zfs.go

```diff
@@ -318,7 +338,7 @@ func (d *zfs) Create() error {
 		d.config["source"] = d.config["zfs.pool_name"]
 	} else {
 		// Validate pool_name.
-		if d.config["zfs.pool_name"] != d.config["source"] {
+		if len(devices) != 1 || d.config["zfs.pool_name"] != devices[0] {
```

Comment:
Ah, so that would have caught some of the issue mentioned above, though I still think flipping the `else if` and `else` sections will make this cleaner.

Reply:
Is this still valid, given the other changes?

Reply:
Given that I'm not convinced the other change solves the problem, I think this is still valid

Reply:
I've rearranged the blocks to align with feedback.